chainio/txmgr/txmgr.go

```diff
@@ -0,0 +1,233 @@
+package txmgr
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/Layr-Labs/eigensdk-go/signerv2"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var (
+	FallbackGasTipCap = big.NewInt(15000000000)
+)
+
+// We are taking inspiration from the optimism TxManager interface
+// https://github.com/ethereum-optimism/optimism/blob/develop/op-service/txmgr/txmgr.go
+
+type TxManager interface {
+	// Send is used to send a transaction
+	// It takes an unsigned transaction and then signs it before sending
+	// It also takes care of nonce management and gas estimation
+	Send(ctx context.Context, tx *types.Transaction) (*types.Receipt, error)
+
+	// Call is used to call a contract method
+	Call(ctx context.Context, msg ethereum.CallMsg, blockNumber *big.Int) ([]byte, error)
+
+	// GetNoSendTxOpts This generates a noSend TransactOpts so that we can use
+	// this to generate the transaction without actually sending it
+	GetNoSendTxOpts() (*bind.TransactOpts, error)
```

Comment:
I'm a bit confused now. Let's talk tomorrow probably easiest. But as a note to my tomorrow-self haha, iiuc txmgr holds the signerFn. If we use a SendTxOpts instead of NoSend, how could it be sent to the chain if its not signed?

Like should this be called GetNoSignNoSendTxOpts()? Or am I completely misunderstanding how abigen works and txs are signed?

Reply:
so the core `transact` function is stupid
https://github.com/ethereum/go-ethereum/blob/master/accounts/abi/bind/base.go#L416

It expects a signer even tho it doesn't want to sign.

Reply:
gotcha. makes sense. Might be good to describe this in the comments even for our future selves.